math/src/elliptic_curve/short_weierstrass/curves/bls12_381/compression.rs

```diff
@@ -106,6 +102,39 @@ impl Compress for BLS12381Curve {
             .ok_or(ByteConversionError::PointNotInSubgroup)
     }
 
+    #[cfg(feature = "alloc")]
+    fn compress_g2_point(point: &Self::G2Point) -> alloc::vec::Vec<u8> {
+        if *point == G2Point::neutral_element() {
+            // point is at infinity
+            let mut x_bytes = alloc::vec![0_u8; 96];
+            x_bytes[0] |= 1 << 7;
+            x_bytes[0] |= 1 << 6;
+            x_bytes
+        } else {
+            // point is not at infinity
+            let point_affine = point.to_affine();
+            let x = point_affine.x();
+            let y = point_affine.y();
+
+            let x_rev: FieldElement<Degree2ExtensionField> =
+                FieldElement::new([x.value()[1].clone(), x.value()[0].clone()]);
+
+            let mut x_bytes = x_rev.to_bytes_be();
+
+            // Set first bit to to 1 indicate this is compressed element.
+            x_bytes[0] |= 1 << 7;
+
+            // Set the 3rd bit based on y value.
+            let y_neg = -y;
+            if y.value()[0].representative() > y_neg.value()[0].representative()
```

Comment:
This condition should be y.value()[0].representative() > y_neg.value()[0].representative() || (y.value()[0].representative() == y_neg.value()[0].representative() && y.value()[1].representative() > y_neg.value()[1].representative()).
Perhaps a more efficient strategy is
y_is_greater_or_equal = y.value()[0].representative() >= y_neg.value()[0].representative().
if y_is_greater_or_equal || (y_is_greater_or_equal && y.value()[1].representative() > y_neg.value()[1].representative()